sawyer_gazebo/src/arm_kinematics_interface.cpp

```diff
@@ -251,33 +251,71 @@ void ArmKinematicsInterface::publishGravityTorques()
   if (joint_state.get())
   {
     intera_core_msgs::SEAJointState gravity_torques;
+    KDL::JntArray jnt_pos, jnt_vel, jnt_eff, jnt_accelerations, jnt_gravity_model, jnt_gravity_only, jnt_zero;
     auto j_state = *joint_state.get();
     auto num_jnts = kinematic_chain_map_[tip_name_].chain.getNrOfJoints();
+    jnt_accelerations.resize(num_jnts);
+    jnt_zero.resize(num_jnts);
+    KDL::SetToZero(jnt_accelerations);
+    KDL::SetToZero(jnt_zero);
     gravity_torques.name.resize(num_jnts);
     gravity_torques.actual_position.resize(num_jnts);
     gravity_torques.actual_velocity.resize(num_jnts);
     gravity_torques.actual_effort.resize(num_jnts);
-    gravity_torques.gravity_only.resize(num_jnts);
-    // FIXME(imcmahon): Populate below fields
     gravity_torques.commanded_position.resize(num_jnts);
     gravity_torques.commanded_velocity.resize(num_jnts);
     gravity_torques.commanded_acceleration.resize(num_jnts);
     gravity_torques.commanded_effort.resize(num_jnts);
+    std::shared_ptr<const intera_core_msgs::JointCommand> joint_command;
+    joint_command_buffer_.get(joint_command);
+    if (joint_command.get()){
+      for (auto i = 0; i<num_jnts; i++) {
+        for(auto j = 0; j < joint_command->names.size(); j++) {
+          if(joint_command->names[j] == kinematic_chain_map_[tip_name_].joint_names[i])
+          {
+            if(joint_command->position.size() == joint_command->names.size() &&
+                 (joint_command->mode == joint_command->POSITION_MODE ||
+                  joint_command->mode == joint_command->TRAJECTORY_MODE))
+              gravity_torques.commanded_position[i] = joint_command->position[j];
+            if(joint_command->velocity.size() == joint_command->names.size() &&
+                 (joint_command->mode == joint_command->VELOCITY_MODE ||
+                  joint_command->mode == joint_command->TRAJECTORY_MODE))
+              gravity_torques.commanded_velocity[i] = joint_command->velocity[j];
+            if(joint_command->effort.size() == joint_command->names.size() &&
+                 joint_command->mode == joint_command->TORQUE_MODE)
+              gravity_torques.commanded_effort[i] = joint_command->effort[j];
+            if(joint_command->acceleration.size() == joint_command->names.size() &&
+                 joint_command->mode == joint_command->TRAJECTORY_MODE)
+            {
+              gravity_torques.commanded_acceleration[i] = joint_command->acceleration[j];
+              if((ros::Time::now() - joint_command->header.stamp).toSec() < 0.2)
```

Comment:
Why is the time checked for acceleration but none of the other joint states? The magic number of 0.2 should also be defined somewhere.